Source/ACE/Network/GameAction/Actions/GameActionFinishBarber.cs

```diff
@@ -13,7 +7,9 @@ public static class GameActionFinishBarber
         [GameAction(GameActionType.FinishBarber)]
         public static void Handle(ClientMessage message, Session session)
         {
-            session.Player.HandleActionFinishBarber(message);
+            ActionChain chain = new ActionChain();
+            chain.AddAction(session.Player, () => session.Player.HandleActionFinishBarber(message));
```

Comment:
I will adjust this, then. Because no one, including myself, will ever go back and do it later! Hold off on pushing, please.